session/bootstrap.go

```diff
@@ -2789,6 +2802,36 @@ func upgradeToVer172(s Session, ver int64) {
 	mustExecute(s, CreateDoneRunawayWatchTable)
 }
 
+func upgradeToVer173(s Session, ver int64) {
+	if ver >= version173 {
+		return
+	}
+	doReentrantDDL(s, `ALTER TABLE mysql.user
+	ADD COLUMN ssl_type enum('','ANY','X509','SPECIFIED') NOT NULL DEFAULT '',
```

Comment:
The column order of mysq.user in TiDB is different from MySQL,
Will we keep it consistent?

Reply:
Tools should avoid using any `mysql.*` tables directly. Instead they should prefer using things like `CREATE USER`, `ALTER USER`, `GRANT`, `REVOKE`, etc.

When that's not possible (e.g. to list users) it might be needed to directly query `mysql.user`. When this is done the query should specify the columns that the tool needs. Using `SELECT * FROM ...` should not be used by applications unless the application is willing to handle different order of columns and more or less columns than expected etc.

We could try to keep this the same, but there are already a lot of differences. If we do this we should do this outside of this PR.

```
sql> SELECT VERSION(); SHOW COLUMNS FROM mysql.user;
+-----------+
| VERSION() |
+-----------+
| 8.1.0     |
+-----------+
1 row in set (0.0007 sec)
+--------------------------+-----------------------------------+------+-----+-----------------------+-------+
| Field                    | Type                              | Null | Key | Default               | Extra |
+--------------------------+-----------------------------------+------+-----+-----------------------+-------+
| Host                     | char(255)                         | NO   | PRI |                       |       |
| User                     | char(32)                          | NO   | PRI |                       |       |
| Select_priv              | enum('N','Y')                     | NO   |     | N                     |       |
| Insert_priv              | enum('N','Y')                     | NO   |     | N                     |       |
| Update_priv              | enum('N','Y')                     | NO   |     | N                     |       |
| Delete_priv              | enum('N','Y')                     | NO   |     | N                     |       |
| Create_priv              | enum('N','Y')                     | NO   |     | N                     |       |
| Drop_priv                | enum('N','Y')                     | NO   |     | N                     |       |
...
```

```
sql> SELECT VERSION(); SHOW COLUMNS FROM mysql.user;
+------------------------------------------+
| VERSION()                                |
+------------------------------------------+
| 5.7.25-TiDB-v7.3.0-alpha-309-g8b8ba8b648 |
+------------------------------------------+
1 row in set (0.0011 sec)
+------------------------+-----------------------------------+------+-----+-------------------+-------+
| Field                  | Type                              | Null | Key | Default           | Extra |
+------------------------+-----------------------------------+------+-----+-------------------+-------+
| Host                   | char(255)                         | NO   | PRI | NULL              |       |
| User                   | char(32)                          | NO   | PRI | NULL              |       |
| authentication_string  | text                              | YES  |     | NULL              |       |
| plugin                 | char(64)                          | YES  |     | NULL              |       |
| Select_priv            | enum('N','Y')                     | NO   |     | N                 |       |
| Insert_priv            | enum('N','Y')                     | NO   |     | N                 |       |
| Update_priv            | enum('N','Y')                     | NO   |     | N                 |       |
| Delete_priv            | enum('N','Y')                     | NO   |     | N                 |       |
| Create_priv            | enum('N','Y')                     | NO   |     | N                 |       |
| Drop_priv              | enum('N','Y')                     | NO   |     | N                 |       |
| Process_priv           | enum('N','Y')                     | NO   |     | N                 |       |
...
```

Note that TiDB also has at least one extra column: `Token_issuer`